Bug: PHPUnit Tests Failing

## Summary

It seems our PHPUnit tests are periodically failing even when no changes to PHP are made.

## Details

```sh
PHP Notice:  Constant FAR_FUTURE_DATE already defined in /__w/gc-digital-talent/gc-digital-talent/api/tests/Feature/UserTest.php on line 20
PHPUnit 9.5.20 #StandWithUkraine
Runtime:       PHP 7.4.30
Configuration: /__w/gc-digital-talent/gc-digital-talent/api/phpunit.xml
....................................F.....                        42 / 42 (100%)
Time: 00:05.201, Memory: 54.50 MB
There was 1 failure:
1) UserTest::testFilterByProfileComplete
Unable to find JSON: 
[{
    "data": {
        "usersPaginated": {
            "paginatorInfo": {
                "total": 3
            }
        }
    }
}]
within response JSON:
[{
    "data": {
        "usersPaginated": {
            "paginatorInfo": {
                "total": 0
            }
        }
    }
}].
Failed asserting that an array has the subset Array &0 (
    'data' => Array &1 (
        'usersPaginated' => Array &2 (
            'paginatorInfo' => Array &3 (
                'total' => 3
            )
        )
    )
).
--- Expected
+++ Actual
@@ @@
     array (
       'paginatorInfo' => 
       array (
-        'total' => 3,
+        'total' => 0,
       ),
     ),
   ),
 )
/__w/gc-digital-talent/gc-digital-talent/api/vendor/laravel/framework/src/Illuminate/Testing/Constraints/ArraySubset.php:[8](https://github.com/GCTC-NTGC/gc-digital-talent/runs/6900470409?check_suite_focus=true#step:6:9)5
/__w/gc-digital-talent/gc-digital-talent/api/vendor/laravel/framework/src/Illuminate/Testing/Assert.php:40
/__w/gc-digital-talent/gc-digital-talent/api/vendor/laravel/framework/src/Illuminate/Testing/AssertableJsonString.php:273
/__w/gc-digital-talent/gc-digital-talent/api/vendor/laravel/framework/src/Illuminate/Testing/TestResponse.php:6[9](https://github.com/GCTC-NTGC/gc-digital-talent/runs/6900470409?check_suite_focus=true#step:6:10)5
/__w/gc-digital-talent/gc-digital-talent/api/tests/Feature/UserTest.php:[12](https://github.com/GCTC-NTGC/gc-digital-talent/runs/6900470409?check_suite_focus=true#step:6:13)84
phpvfscomposer:///__w/gc-digital-talent/gc-digital-talent/api/vendor/phpunit/phpunit/phpunit:97
FAILURES!
Tests: 42, Assertions: [14](https://github.com/GCTC-NTGC/gc-digital-talent/runs/6900470409?check_suite_focus=true#step:6:15)3, Failures: 1.
Error: Process completed with exit code 1.
```

--- a/api/database/factories/UserFactory.php
+++ b/api/database/factories/UserFactory.php
@@ -37,7 +37,7 @@ class UserFactory extends Factory
         return [
             'first_name' => $this->faker->firstName(),
             'last_name' => $this->faker->lastName(),
-            'email' => $this->faker->boolean(75) ? $this->faker->unique()->safeEmail : null,
+            'email' => $this->faker->unique()->safeEmail(),
             'sub' => $this->faker->boolean(75) ? $this->faker->unique()->uuid() : null,
             'telephone' => $this->faker->e164PhoneNumber(),
             'preferred_lang' => $this->faker->randomElement(['en', 'fr']),

--- a/api/database/seeders/DatabaseSeeder.php
+++ b/api/database/seeders/DatabaseSeeder.php
@@ -16,6 +16,7 @@ use Illuminate\Support\Facades\DB;
 use App\Models\AwardExperience;
 use App\Models\CommunityExperience;
 use App\Models\EducationExperience;
+use App\Models\GenericJobTitle;
 use App\Models\PersonalExperience;
 use App\Models\WorkExperience;
 use Faker;
@@ -61,6 +62,9 @@ class DatabaseSeeder extends Seeder
             ->afterCreating(function (User $user) use ($faker) {
                 $assets = CmoAsset::inRandomOrder()->limit(4)->pluck('id')->toArray();
                 $user->cmoAssets()->sync($assets);
+
+                $genericJobTitles = GenericJobTitle::inRandomOrder()->limit(2)->pluck('id')->toArray();
+                $user->expectedGenericJobTitles()->sync($genericJobTitles);
 
                 // pick a pool in which to place this user
                 $pool = Pool::inRandomOrder()->limit(1)->first();

--- a/api/tests/Feature/UserTest.php
+++ b/api/tests/Feature/UserTest.php
@@ -10,11 +10,14 @@ use App\Models\CommunityExperience;
 use App\Models\EducationExperience;
 use App\Models\PersonalExperience;
 use App\Models\WorkExperience;
+use App\Models\GenericJobTitle;
 use Illuminate\Foundation\Testing\RefreshDatabase;
 use Nuwave\Lighthouse\Testing\ClearsSchemaCache;
 use Nuwave\Lighthouse\Testing\MakesGraphQLRequests;
 use Tests\TestCase;
 use Database\Helpers\ApiEnums;
+use Database\Seeders\ClassificationSeeder;
+use Database\Seeders\GenericJobTitleSeeder;
 
 // for setting expiry dates so tests don't fail due to expired candidates when they are not supposed to
 const FAR_FUTURE_DATE = '2050-01-01';
@@ -1307,6 +1310,10 @@ class UserTest extends TestCase
 
     public function testFilterByProfileComplete(): void
     {
+        // need some generic job titles for a complete profile
+        $this->seed(ClassificationSeeder::class);
+        $this->seed(GenericJobTitleSeeder::class);
+
         // Create initial set of 5 users with incomplete profiles.
         User::factory()->count(5)->create([
             'current_province' => null,
@@ -1333,18 +1340,22 @@ class UserTest extends TestCase
         ]);
 
         // Create some complete users.
-        User::factory()->count(3)->create([
-            'current_province' => 'ONTARIO',
-            'location_preferences' => ['PRAIRIE'],
-            'looking_for_english' => null,
-            'looking_for_french' => true,
-            'looking_for_bilingual' => null,
-            'telephone' => '+15407608748',
-            'current_city' => 'Somewhere random',
-            'is_gov_employee' => false,
-            'expected_salary' => ['_50_59K'],
-            'would_accept_temporary' => false,
-        ]);
+        User::factory()->count(3)
+            ->afterCreating(function($user) {
+                $user->expectedGenericJobTitles()->sync([GenericJobTitle::first()->id]);
+            })
+            ->create([
+                'current_province' => 'ONTARIO',
+                'location_preferences' => ['PRAIRIE'],
+                'looking_for_english' => null,
+                'looking_for_french' => true,
+                'looking_for_bilingual' => null,
+                'telephone' => '+15407608748',
+                'current_city' => 'Somewhere random',
+                'is_gov_employee' => false,
+                'expected_salary' => ['_50_59K'],
+                'would_accept_temporary' => false,
+            ]);
 
         // Assert query no isProfileComplete filter will return all users
         $this->graphQL(/** @lang Graphql */ '

